### PR TITLE
Review comments from previous PR and add browser language detection

### LINKED ISSUE #2: Make emails available in different languages

### LINKED ISSUE #3: Get templates from external source
The idea would be to be able to get templates from external modules. 
It has to work with multiple languages.

### LINKED ISSUE #4: Manage templates with different content
We would like to manage templates using different content/layout.
The idea is to have a new environment variable that will define which templates will be loaded at startup.

### LINKED ISSUE #8: Refactor i18n code
Follow recommendations and comments defined in PR #7  to refactor the code

### PR DESCRIPTION
**Pull request by BigFab**
This PR includes code review based on comments from previous PRs.
Also it is adding some code to get the user browser language that will be used for emailing in case the user profile is not found (for example, when the user is requesting for password reset, he is not logged in hence there is no preferred language)

fix #8 
fix #7
fix #2 
fix #3 
fix #4

### CODE REVIEW

File: api/forgot.go [hunk 1]
```diff
@@ -51,8 +57,20 @@ func (a *Api) passwordReset(res http.ResponseWriter, req *http.Request, vars map
 	resetCnf, _ := models.NewConfirmation(models.TypePasswordReset, models.TemplateNamePasswordReset, "")
 	resetCnf.Email = email
 
+	// if the reseter is already a Tidepool user, we can use his preferences
 	if resetUsr := a.findExistingUser(resetCnf.Email, a.sl.TokenProvide()); resetUsr != nil {
 		resetCnf.UserId = resetUsr.UserID
+
+		// let's get the reseter user preferences
+		reseterPreferences := &models.Preferences{}
+		if err := a.seagull.GetCollection(resetCnf.UserId, "preferences", a.sl.TokenProvide(), reseterPreferences); err != nil {
```
Comment (ydrisR) [inline api/forgot.go:66]: Just a coding style remark: would be nice if we can avoid long lines like this. 
I don't want to be too restrictive but lines longer than 100 characters are definitely difficult to read ;)

File: api/forgot.go [hunk 2]
```diff
@@ -51,8 +57,20 @@ func (a *Api) passwordReset(res http.ResponseWriter, req *http.Request, vars map
...
+			a.sendError(res, http.StatusInternalServerError, STATUS_ERR_FINDING_USR, "forgot password: error getting reseter user preferences: ", err.Error())
```
Comment (ydrisR) [inline api/forgot.go]: same remark as above

File: api/internationalization.go [hunk 3]
```diff
@@ -147,3 +140,44 @@ func getAllLocalizationFiles(templatesPath string) ([]string, error) {
 	}
 	return retFiles, nil
 }
+
+//getBrowserPreferredLanguage returns the preferred language extracted from the request browser
+func getBrowserPreferredLanguage(req *http.Request) string {
+
+	if acptlng := req.Header.Get(HEADER_LANGUAGE); acptlng == "" {
+		return ""
+	} else if langs := parseAcceptLanguage(acptlng); langs == nil {
```
Comment (ydrisR) [inline api/internationalization.go]: Not sure "langs" exists in english

File: api/invite.go [hunk 4]
```diff
@@ -399,9 +402,21 @@ func (a *Api) SendInvite(res http.ResponseWriter, req *http.Request, vars map[st
 			//None exist so lets create the invite
 			invite, _ := models.NewConfirmationWithContext(models.TypeCareteamInvite, models.TemplateNameCareteamInvite, invitorID, ib.Permissions)
 
+			// if the invitee is already a Tidepool user, we can use his preferences
 			invite.Email = ib.Email
 			if invitedUsr != nil {
 				invite.UserId = invitedUsr.UserID
+
+				// let's get the invitee user preferences
+				inviteePreferences := &models.Preferences{}
```
Comment (ydrisR) [inline api/invite.go:411]: Looks like a block of code (411 - 419) common between invite.go and forgot.go. 
Isn't it something worth to factorize?

File: env.sh [hunk 5]
```diff
@@ -31,7 +31,7 @@ export TIDEPOOL_HYDROPHONE_SERVICE='{
         "serverSecret": "This needs to be the same secret everywhere. YaHut75NsK1f9UKUXuWqxNN0RUwHFBCy",
         "webUrl": "http://localhost:3000",
         "assetUrl": "https://s3-us-west-2.amazonaws.com/tidepool-dev-asset",
-        "internationalizationTemplatesPath": "/go/src/github.com/tidepool-org/hydrophone/templates"
+        "i18nTemplatesPath": "/go/src/github.com/tidepool-org/hydrophone/templates"
```
Comment (ydrisR) [inline env.sh:34]: Please create PR on the config folder to reflect this change and link both PRs together

File: templates/templates.go [hunk 6]
```diff
@@ -11,97 +11,107 @@ import (
...
 	} else {
 		templates[template.Name()] = template
 	}
 
 	return templates, nil
 }
+
+//NewGenericTemplate returns the requested template
+//templateName is the name of the template to be returned
+func NewGenericTemplate(templatesPath string, templateName models.TemplateName) (models.Template, error) {
```
Comment (ydrisR) [inline templates/templates.go]: Would change the method name to "NewTemplate". This function returns a specific template (according to the template name provided in the args), not a generic template. This is missleading

**Review by ydrisR — CHANGES_REQUESTED** [review]
Looks good! 👍 
Just a few code style and naming remarks

File: templates/templates.go (hunk 6 above)
Comment (BigFab) [reply templates/templates.go]: OK

File: api/internationalization.go (hunk 3 above)
Comment (BigFab) [reply api/internationalization.go]: moved to "languages" ^^

File: api/forgot.go (hunk 1 above)
Comment (BigFab) [reply api/forgot.go:66]: Go parser here is cancelling my tentatives of wrapping this line

File: api/forgot.go (hunk 2 above)
Comment (BigFab) [reply api/forgot.go]: this can be worked out by wrapping after each comma

File: api/invite.go (hunk 4 above)
Comment (BigFab) [reply api/invite.go:411]: I think we could have a discussion on this one but at the bottom I dont think it is worth investigating a possible refactor here

File: env.sh (hunk 5 above)
Comment (BigFab) [reply env.sh:34]: done: https://github.com/mdblp/configs/pull/56

**Review by ydrisR — APPROVED** [review]

**Pull request merged**